server/sonar-server/src/main/java/org/sonar/server/computation/metric/MetricRepositoryImpl.java

```diff
@@ -19,47 +19,91 @@
  */
 package org.sonar.server.computation.metric;
 
+import com.google.common.base.Function;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import org.picocontainer.Startable;
 import org.sonar.core.metric.db.MetricDto;
 import org.sonar.core.persistence.DbSession;
 import org.sonar.server.db.DbClient;
 
+import static com.google.common.collect.FluentIterable.from;
 import static java.util.Objects.requireNonNull;
 
-public class MetricRepositoryImpl implements MetricRepository {
+public class MetricRepositoryImpl implements MetricRepository, Startable {
   private final DbClient dbClient;
+  @CheckForNull
+  private Map<String, Metric> metricsByKey;
+  @CheckForNull
+  private Map<Long, Metric> metricsById;
 
   public MetricRepositoryImpl(DbClient dbClient) {
     this.dbClient = dbClient;
   }
 
   @Override
+  public void start() {
+    try (DbSession dbSession = dbClient.openSession(false)) {
```

Comment:
I know that the Jacoco issue is not our responsability, but using a try-with-resource will drop the branch coverage of this class...